week2/index.js

```diff
@@ -0,0 +1,163 @@
+"use-strict";
+
+const fs = require('fs');
+const TODO_FILE = process.env['TODO_file'] || 'todo.json'
+const HELP_MENU = 'help.txt';
+
+let readFiles = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, "utf8", (err, data) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    // No such file or directory
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    // the user don't have a permission
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    // any other Error
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                process.exit();
+            }
+            resolve(data);
+
+        })
+    })
+
+};
+
+let writeFiles = (filename, dataAsString) => {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(TODO_FILE, dataAsString, (err) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    // No such file or directory
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    // the user don't have a permission
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    // any other Error
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                process.exit();

```

Comment:
If we exit the process during a failure here, we might actually lose data (the new todo we were trying to enter), and data loss is just about the worst sin that code can commit.

Like above, i think it might make more sense to reject the promise, and let the caller retry or do something else.